### PR TITLE
Proposed fix for #1532 Tracks in song editor cannot be moved in pattern ...

### PR DESCRIPTION
**Pull request by curlymorphic**
Proposed fix for #1532 Tracks in song editor cannot be moved in pattern ...

### CODE REVIEW

**Comment by Sti2nd** [comment]
Neat! That is the case with Piano Roll too.. hint hint :)

I am curios how you make a new selection. You have to click on an empty area to wipe the old one out?

**Comment by curlymorphic** [comment]
> I am curios how you make a new selection. You have to click on an empty area to wipe the old one out?

as soon as you press the mouse in select mode, the current selection is cleared

**Comment by badosu** [comment]
@curlymorphic Lol, I was tinkering with this issue some days ago, I am glad you were able to fix this before I put much effort into a duplicate. Thanks!

**Comment by badosu** [comment]
I would like to make a suggestion though: commit messages are very important to track history and functionality changes, bug tracking, etc...

I noticed that most of your commit messages are "Proposed fix xxx: ..." Where xxx is the issue number and the beggining of the issue title.

It would help a lot if you could use descriptive messages of what the commit does instead, for example: "Fix tracks can't be moved when in pattern selection mode".

Git does not knows exactly about github and should be agnostic of bug tracker, even thougn we can always use the description area of the commit to links this information: "Fix #1532". Github uses this as an actual way to figure out which issue you are referring to and closes the issue automatically when the PR is merged.

See https://github.com/erlang/otp/wiki/Writing-good-commit-messages, this is a nice guideline.

If you don't agree with this idea it's fine as well. :-). Thanks for everything!

File: src/core/Track.cpp [hunk 1]
```diff
@@ -2612,7 +2612,11 @@ void TrackView::mousePressEvent( QMouseEvent * _me )
 	}
 
 
-	if( m_trackContainerView->allowRubberband() == true )
+	int widgetTotal = ConfigManager::inst()->value( "ui",
```
Comment (badosu) [inline src/core/Track.cpp]: This is the width of the left part of the track view, with all the buttons and leds and stuff right? Good job! :-)

**Comment by Sti2nd** [comment]
> as soon as you press the mouse in select mode, the current selection is cleared

ahh, that still works. Not always easy to think how a feature works.

**Comment by tresf** [comment]
> It would help a lot if you could use descriptive messages of what the commit does instead, for example: "Fix tracks can't be moved when in pattern selection mode".

@badosu I too have found myself trying to locate the purpose of a PR by finding the parent issue.  I personally don't like the `#1532` hash in the title because it double-links the title.  It's like writing "[Click here to visit [`#google`](http://yahoo.com)](http://google.com)"

So because of that, I tend to put "Closes `#1532`" in the body so that the tile link remains intact.  Does that make sense?

**Comment by badosu** [comment]
@tresf Sorry, I did not understand your last comment. Isn't it the same thing I said? BTW: Fixes and Closes is tracked by github on merge as well, see https://help.github.com/articles/closing-issues-via-commit-messages/

**Comment by tresf** [comment]
> Isn't it the same thing I said? 

Yes absolutely however I wanted to point out the double linking issue as well.

**Comment by tresf** [comment]
In other words, not putting the hash mark in the title of the pull request is preferred in my opinion.  That "Dave-ism" I tend to agree with.

**Comment by curlymorphic** [comment]
I did reply to this earlier, then forget to commit it :(

so are we saying we dont want the "Proposed Fix 1234" at the start of my commit message?

**Comment by badosu** [comment]
@curlymorphic Well, it's just an idea, it's fine by me if it's important for you to mantain Dave's style. :-)

It's just that the guideline I shared is kind of an established ideal in the industry, and that's for a good reason: helps a lot when you use [git bisect](http://git-scm.com/docs/git-bisect), [git blame](http://git-scm.com/docs/git-blame), searching for commits that touched a functionality by the title, etc...

For example, a suggestion for the current case would be:

```
Fix: track can't be moved in selection mode (brief description of what de commit does)

Blablabla (reason why this commit is important, important concerns, history on it, etc)

Fixes #1521 (References github and automatically closes the issue when merged to master branch)
```

**Comment by badosu** [comment]
For example, this is the list of changes from release 1.1.0 to 1.1.1:

```
~/C/lmms ❯❯❯ git log v1.1.0...v1.1.1 --no-merges --oneline
523e5d0 Bump version for stable-1.1 patch release
e33645f Fix monstro slopes
49b3f36 1492 fixed type on line 313
a1dce66 Proposed fix 1492 Nescaline crash when playing very high note
ffde891 Proposed fix 1526 Watsyn shows wrong Osc
44f72b0 Added const
93b6888 Fix export for some locales
5bf095b Revert Zynaddsubfx changes
62df768 Autoquit improvement: On effect plugins where it's possible for the FX to silence the output, measure the levels of the input signal for autoquit This so that the effect won't be turned off when there's input that the effect is muting (eg. when you use the Amplifier to temporarily mute a signal)
b5538c7 used newWaveView in constructor, to remove redundancy
76a1b8c Proposed fix for 929 AFP wave display
36d02b9 Fix memory leaks in Audio setup tab in the settings dialog
e0f7ea5 Check if object already exists instead of deleting and recreating it
0a732fb Fix mismatched free() / delete / delete [] in RmsHelper.h
b59a501 Fix memory leaks in ZynAddSubFX
ddbb180 Fix memory leaks in LADSPA plugins
```

It's still manageable, but try 1.1.0 to master and you'll understand what I am saying. :-)

File: src/core/Track.cpp (hunk 1 above)
Comment (curlymorphic) [inline src/core/Track.cpp]: I cant really take credit for a copy and paste :)

**Comment by curlymorphic** [comment]
> if it's important for you to maintain Dave's style.

no not really. More important for me to get it right.

**Pull request merged**